services/core/shared/core/compat/sdk_v5/blocksUtils.js

```diff
@@ -92,23 +93,8 @@ const downloadGenesisBlock = async () => {
 	logger.info(`Downloading genesis block to the filesystem from: ${genesisBlockUrl}`);
 
 	return new Promise((resolve, reject) => {
-		if (genesisBlockUrl.endsWith('.tar.gz')) {
-			https.get(genesisBlockUrl, (response) => {
-				if (response.statusCode === 200) {
-					response.pipe(tar.extract({ cwd: directoryPath }));
-					response.on('error', async (err) => reject(err));
-					response.on('end', async () => {
-						logger.info('Genesis block downloaded successfully');
-						return setTimeout(resolve, 500);
-					});
-				} else {
-					const errMessage = `Download failed with HTTP status code: ${response.statusCode} (${response.statusMessage})`;
-					logger.error(errMessage);
-					if (response.statusCode === 404) throw new NotFoundException(errMessage);
-					throw new Error(errMessage);
-				}
-			});
-		} else {
+		if (genesisBlockUrl.endsWith('.tar.gz')) downloadZip(genesisBlockUrl, directoryPath);
+		else {
 			request(genesisBlockUrl)
```

Comment:
Please consider moving this utility as `downloadJSONFile` (or something similar) to `services/core/shared/downloadFile.js` as well.